frechet_audio_distance/fad.py

```diff
@@ -30,38 +33,55 @@ def load_audio_task(fname, dtype="float32"):
         wav_data = wav_data / 32768.0
     elif dtype == 'int32':
         wav_data = wav_data / float(2**31)
-    
+
     # Convert to mono
     if len(wav_data.shape) > 1:
         wav_data = np.mean(wav_data, axis=1)
 
-    if sr != SAMPLE_RATE:
-        wav_data = resampy.resample(wav_data, sr, SAMPLE_RATE)
+    if sr != sample_rate:
+        wav_data = resampy.resample(wav_data, sr, sample_rate)
 
     return wav_data
 
 
 class FrechetAudioDistance:
     def __init__(
-        self, 
+        self,
         ckpt_dir=None,
-        model_name="vggish", 
+        model_name="vggish",
+        submodel_name="630k-audioset",  # only for CLAP

```

Comment:
This might introduce too many optional arguments for the instantiation of `FrechetAudioDistance`. While it is clear with comments right now, it might get bloated if we add in more models that need different arguments in future.

Moving forward, we probably need to introduce config files for different embedding models.
Let's leave it as is for now, and add a note here for future code refactoring.

Reply:
I guess once we also add OpenL3 we will have to do some refactoring to make `FrechetAudioDistance` easy to understand.

Reply:
Yes. We might need to define some data classes for different models, one example might be the different types of model configs in huggingface's `transformers`, and we can keep them as JSON files. 
If you have better ideas feel free to suggest / contribute too! :)